cli/src/cli.rs

```diff
@@ -1458,6 +1440,74 @@ fn process_transfer(
     }
 }
 
+fn process_transfer_many(
+    rpc_client: &RpcClient,
+    config: &CliConfig,
+    lamports: u64,
+    recipients: &[Pubkey],
+    from: SignerIndex,
+    nonce_account: Option<&Pubkey>,
+    nonce_authority: SignerIndex,
+    fee_payer: SignerIndex,
+) -> ProcessResult {
+    let from = config.signers[from];
+    let nonce_authority = config.signers[nonce_authority];
+    let fee_payer = config.signers[fee_payer];
+
+    // Build transactions
+    let (blockhash, fee_calculator, _) = rpc_client
+        .get_recent_blockhash_with_commitment(config.commitment)?
+        .value;
+    let mut transactions: Vec<Transaction> = Vec::new();
+    for to in recipients.iter() {
+        let ixs = vec![system_instruction::transfer(&from.pubkey(), to, lamports)];
+
+        let message = if let Some(nonce_account) = &nonce_account {
+            Message::new_with_nonce(
+                ixs,
+                Some(&fee_payer.pubkey()),
+                nonce_account,
+                &nonce_authority.pubkey(),
+            )
+        } else {
+            Message::new(&ixs, Some(&fee_payer.pubkey()))
+        };
+        transactions.push(Transaction::new_unsigned(message));
+    }
+    let mut message_refs = vec![];
+    for transaction in transactions.iter() {
+        message_refs.push(&transaction.message);
+    }
+
+    check_account_balances(
+        rpc_client,
+        lamports,
+        &fee_calculator,
+        &from.pubkey(),
+        &fee_payer.pubkey(),
+        &message_refs,
+        config.commitment,
+    )?;
+
+    if let Some(nonce_account) = &nonce_account {
+        let nonce_account =
+            nonce::get_account_with_commitment(rpc_client, nonce_account, config.commitment)?;
+        check_nonce_account(&nonce_account, &nonce_authority.pubkey(), &blockhash)?;
+    }
+
+    let results = rpc_client.transact(
+        transactions,
+        &config.signers,
+        &fee_payer.pubkey(),
+        TransactConfig {
+            spinner: Some(new_spinner_progress_bar()),
+            ..TransactConfig::default()
+        },
+    )?;
+    let results: CliTransferManyResults = results.into();
```

Comment:
At the end of transfer-many, I think it would be really nice if any permanent failures were written back out to a .csv file.  This makes it easy for the (likely less technical) airdrop operator to retry the airdrop with less risk sending 2x to some users.   

Alternatively I believe @garious's work in stake-accounts has a more sophisticated database mechanism to avoid double sends, could that be repurposed here?